src/js/components/Select/Select.js

```diff
@@ -152,9 +152,33 @@ const Select = forwardRef(
       if (onClose) onClose();
     }, [onClose]);
 
+    const triggerChangeEvent = useCallback(nextValue => {
+      // Calling set value function directly on input because React library
+      // overrides setter `event.target.value =` and loses original event
+      // target fidelity.
+      // https://stackoverflow.com/a/46012210
+      const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+        window.HTMLInputElement.prototype,
+        'value',
+      ).set;
+      nativeInputValueSetter.call(inputRef.current, nextValue);
+      const event = new Event('input', { bubbles: true });
+      inputRef.current.dispatchEvent(event);
+    }, []);
+
     const onSelectChange = useCallback(
       (event, { option, value: nextValue, selected: nextSelected }) => {
         if (closeOnChange) onRequestClose();
+        // nextValue must not be of type object to set value directly on the
+        // input. if it is an object, then the user has not provided necessary
+        // props to reduce object option
+        if (typeof nextValue !== 'object' && nextValue !== event.target.value) {
```

Comment:
Seems that this does not cover the select capability of having options as objects. 
We should take into account if (typeof nextValue === 'object')  and compare define actualValue = nextValue[valueKey] and then compare it to previous